lib/librte_hash/rte_cuckoo_hash.c

```diff
@@ -1160,3 +1160,81 @@ rte_hash_iterate(const struct rte_hash *h, const void **key, void **data, uint32
 
 	return position - 1;
 }
+
+/* Get the primary bucket index given the precomputed hash value. */
+static inline uint32_t
+rte_hash_get_primary_bucket(const struct rte_hash *h, hash_sig_t sig)
+{
+	return sig & h->bucket_bitmask;
+}
+
+/* Get the secondary bucket index given the precomputed hash value. */
+static inline uint32_t
+rte_hash_get_secondary_bucket(const struct rte_hash *h, hash_sig_t sig)
+{
+	return rte_hash_secondary_hash(sig) & h->bucket_bitmask;
+}
+
+static inline void
+rte_bucket_iterator_state_update(struct rte_bucket_iterator_state *state)
+{
+	state->bidx = (state->vnext < RTE_HASH_BUCKET_ENTRIES) ?
+		rte_hash_get_primary_bucket(state->h, state->sig) :
+		rte_hash_get_secondary_bucket(state->h, state->sig);
```

Comment:
We are recomputing the hash every time we advance `state->vnext`. We can save the primary and secondary buckets and then drop field `state->sig`.